Content.Server/_EE/Supermatter/Systems/SupermatterSystem.Processing.cs

```diff
@@ -0,0 +1,615 @@
+
+        var h2OBonus = 1 - gases[Gas.WaterVapor] * 0.25f;
+
+        powerRatio = Math.Clamp(powerRatio, 0, 1);
+        heatModifier = Math.Max(heatModifier, 0.5f);
+        transmissionBonus *= h2OBonus;
+
+        // Effects the damage heat does to the crystal
+        sm.DynamicHeatResistance = 1f;
+
+        // More moles of gases are harder to heat than fewer, so let's scale heat damage around them
+        sm.MoleHeatPenaltyThreshold = (float) Math.Max(moles * sm.MoleHeatPenalty, 0.25);
+
+        // Ramps up or down in increments of 0.02 up to the proportion of CO2
+        // Given infinite time, powerloss_dynamic_scaling = co2comp
+        // Some value from 0-1
+        if (moles > sm.PowerlossInhibitionMoleThreshold && gases[Gas.CarbonDioxide] > sm.PowerlossInhibitionGasThreshold)
+        {
+            var co2powerloss = Math.Clamp(gases[Gas.CarbonDioxide] - sm.PowerlossDynamicScaling, -0.02f, 0.02f);
+            sm.PowerlossDynamicScaling = Math.Clamp(sm.PowerlossDynamicScaling + co2powerloss, 0f, 1f);
+        }
+        else
+            sm.PowerlossDynamicScaling = Math.Clamp(sm.PowerlossDynamicScaling - 0.05f, 0f, 1f);
+
+        // Ranges from 0~1(1 - (0~1 * 1~(1.5 * (mol / 500))))
+        // We take the mol count, and scale it to be our inhibitor
+        var powerlossInhibitor =
+            Math.Clamp(
+                1
+                - sm.PowerlossDynamicScaling
+                * Math.Clamp(
+                    moles / sm.PowerlossInhibitionMoleBoostThreshold,
+                    1f, 1.5f),
+                0f, 1f);
+
+        if (sm.MatterPower != 0) // We base our removed power off 1/10 the matter_power.
+        {
+            var removedMatter = Math.Max(sm.MatterPower / sm.MatterPowerConversion, 40);
+            // Adds at least 40 power
+            sm.Power = Math.Max(sm.Power + removedMatter, 0);
+            // Removes at least 40 matter power
+            sm.MatterPower = Math.Max(sm.MatterPower - removedMatter, 0);
+        }
+
+        // Based on gas mix, makes the power more based on heat or less effected by heat
+        var tempFactor = powerRatio > 0.8 ? 50f : 30f;
+
+        // If there is more pluox and N2 then anything else, we receive no power increase from heat
+        sm.Power = Math.Max(absorbedGas.Temperature * tempFactor / Atmospherics.T0C * powerRatio + sm.Power, 0);
+
+        // Irradiate stuff
+        if (TryComp<RadiationSourceComponent>(uid, out var rad))
+        {
+            rad.Intensity =
+                _config.GetCVar(ECCVars.SupermatterRadsBase) +
+                (sm.Power
+                * Math.Max(0, 1f + transmissionBonus / 10f)
+                * 0.003f
+                * _config.GetCVar(ECCVars.SupermatterRadsModifier));
+
+            rad.Slope = Math.Clamp(rad.Intensity / 15, 0.2f, 1f);
+        }
+
+        // Power * 0.55 * 0.8~1
+        var energy = sm.Power * sm.ReactionPowerModifier;
```

Comment:
```suggestion
        // Power * 0.55 * 0.8~1
        // This has to be differentiated with respect to time, since its going to be interacting with systems
        // that also differentiate. Basically, if we don't multiply by 2 * frameTime, the supermatter will explode faster if your server's tickrate is higher.
        var energy = 2 * sm.Power * sm.ReactionPowerModifier * frameTime;
```